include/deal.II/fe/mapping_q_internal.h

```diff
@@ -2050,16 +2060,25 @@ namespace internal
             CellSimilarity::none,
             data,
             output_data.quadrature_points,
+            output_data.jacobians,
+            output_data.inverse_jacobians,
             output_data.jacobian_grads);
         }
       else
         {
-          maybe_compute_q_points<dim, spacedim>(data_set,
-                                                data,
-                                                output_data.quadrature_points);
-          maybe_update_Jacobians<dim, spacedim>(CellSimilarity::none,
-                                                data_set,
-                                                data);
+          auto quadrature_dim = QProjector<dim>::project_to_face(
+            ReferenceCells::get_hypercube<dim>(), quadrature, face_no);
```

Comment:
This is new!? Where is `subface_no` used!?

Reply:
Also, I don't think this is right. You are not using anywhere `data_set` (and `data`). Latter contains the actual support points. See 

https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/source/fe/mapping_q.cc#L913-L916

https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/source/fe/mapping_q.cc#L219

OK. It does not: 

https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/source/fe/mapping_q.cc#L84

https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/include/deal.II/fe/mapping_q.h#L394-L554

It only stores the shape functions evaluated (used here https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/include/deal.II/fe/mapping_q_internal.h#L1304-L1314; but you need the offset!!!). But adding the quadrature points should not be a problem.

_Note: this reminds me of horrible times when simplex/wedges/pyramids had to be made working for mapping. That the face quadratures are stored in a single vector and you need to use the offsets took me some time...._

Reply:
Regarding this comment, I think we should also remove the variables https://github.com/dealii/dealii/blob/6288d4de9aee5e439f8991f0e9a905d82a86c7ba/include/deal.II/fe/mapping_q.h#L394-L414 to ensure that we really do not use the values, first or second derivatives in a tabulated form any more.

Reply:
@kronbichler Sounds good :+1: